core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalServiceAction.kt

```diff
@@ -0,0 +1,125 @@
+package org.evomaster.core.problem.external.service
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.*
+import com.github.tomakehurst.wiremock.common.Metadata.metadata
+import org.evomaster.core.problem.external.service.param.ResponseParam
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.StructuralElement
+import org.evomaster.core.search.gene.Gene
+import org.evomaster.core.search.gene.GeneUtils
+
+/**
+ * Action to execute the external service related need
+ * to handle the external service calls.
+ *
+ * Typically, handle WireMock responses
+ */
+class ExternalServiceAction(
+
+    /**
+     * Received request to the respective WireMock instance
+     *
+     * TODO: Need to expand the properties further in future
+     *  depending on the need
+     */
+    val request: ExternalServiceRequest,
+
+    /**
+     * currently, we support response with json format
+     * then use ObjectGene now,
+     * might extend it later
+     */
+    val response: ResponseParam = ResponseParam(),
+
+    /**
+     * WireMock server which received the request
+     */
+    val wireMockServer: WireMockServer,
+    private val id: Long,
+
+    val representExistingRequest: Boolean = false
+) : Action(listOf(response)) {
+
+    companion object {
+        private fun buildResponse(template: String): ResponseParam {
+            // TODO: refactor later
+            return ResponseParam()
+        }
+    }
+
+    constructor(request: ExternalServiceRequest, template: String, wireMockServer: WireMockServer, id: Long) :
+            this(request, buildResponse(template), wireMockServer, id)
+
+    init {
+        // TODO: This is not the correct way to do this, but for now
+        //  to test concept, this is triggered here.
+        this.buildResponse()
+    }
+
+    /**
+     * UUID generated by WireMock is used under ExternalServiceRequest
+     * is used as ID for action.
+     *
+     * TODO: After the ID refactor, this needs to be changed.
+     */
+    override fun getName(): String {
+        return request.getID()
+    }
+
+    override fun seeGenes(): List<out Gene> {
+        return response.genes
+    }
+
+    override fun shouldCountForFitnessEvaluations(): Boolean {
+        return false
+    }
+
+    override fun copyContent(): StructuralElement {
+        return ExternalServiceAction(
+            request,
+            response.copy() as ResponseParam,
+            wireMockServer,
```

Comment:
hmmmm... so copied actions will share the same wiremock instance? is it intended? if yes, explain why here in a code comment. if not, refactor

Reply:
Each external service will have a WireMock instance representing that so when the ExternalServiceAction is copied, same instance will be passed into the copy too. Otherwise, we have to manage multiple instances for the same external service.